packages/openapi3-parser/lib/parser.js

```diff
@@ -1,22 +1,139 @@
+/* eslint-disable no-underscore-dangle */
+
 const R = require('ramda');
 const parseYAML = require('./parser/parseYAML');
 
-const { isAnnotation, isObject } = require('./predicates');
+const { isAnnotation, isWarningAnnotation, isObject } = require('./predicates');
 const { createError } = require('./elements');
 const pipeParseResult = require('./pipeParseResult');
 
 const parseOpenAPIObject = require('./parser/oas/parseOpenAPIObject');
 
 const isObjectOrAnnotation = R.either(isObject, isAnnotation);
 
+const recurseSkippingAnnotations = R.curry((visitor, e) => {
```

Comment:
I realised that I was adding logic into `parseOpenAPIObject` because similiar stuff was done there. Although it doesn't belong there, the first commit in this PR is transfering existing logic from happening in the OpenAPI Object parser, to instead be in the generic parsing logic. These pieces of code are not specific to OpenAPI Object parsing, just that was an integration point that contained most of the parse result.